buffer/src/main/java/io/netty/buffer/ByteBufAllocatorMetricProvider.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer;
+
+public interface ByteBufAllocatorMetricProvider {
```

Comment:
I'm less concerned about where the methods are implemented (on the allocator, not on the allocator) and more concerned about having a well defined interface for them. That way folks should be able to limit access to what they pass around, have a well defined interface (which may allow for easier mocking/testing), and users to respect principle of least privilege.